src/cpp/session/modules/SessionRCompletions.R

```diff
@@ -3323,6 +3323,7 @@ assign(x = ".rs.acCompletionTypes",
    .rs.makeCompletions(
       token = token,
       results = results.sorted,
-      packages = packages.sorted
+      packages = packages.sorted,
+      type = attr(results, "type")
```

Comment:
That's a good point; I think you're right.

Also note that one other underlying issue here is that the completion system only gets access to the current line, whereas a full completion system likely needs to know the whole document context (e.g. if you're requesting argument completions on a new line).

FWIW, RStudio's current completion system does a lot of the completion context generation on the front-end, and then later generates completions server-side using that context. I think it would likely be better (somewhere down the road) to completely eschew the client-side context creation and just do everything server-side.

In other words, we might want to avoid coupling `completeme` to the current state of the RStudio completion system, and instead consider how we might properly re-design this (and later just re-write RStudio to appropriate hook in to `completeme`).

What are your thoughts?